File: arven-towns-api/src/main/java/arvenwood/towns/api/claim/ClaimService.java
package arvenwood.towns.api.claim;

import arvenwood.towns.api.town.Town;
import org.spongepowered.api.Sponge;
import org.spongepowered.api.world.Location;
import org.spongepowered.api.world.World;

import java.util.Collection;
import java.util.Optional;

public interface ClaimService {

    static ClaimService get() {
        return Sponge.getServiceManager().provideUnchecked(ClaimService.class);
    }

    Optional<Claim> getClaimAt(Location<World> location);

    Collection<Claim> getClaimsFor(World world);

    Collection<Claim> getClaimsFor(Town town);

    boolean contains(Claim claim);

    boolean register(Claim claim);

    boolean unregister(Claim claim);
}

File: src/main/kotlin/arvenwood/towns/api/claim/ClaimService.kt
package arvenwood.towns.api.claim

import arvenwood.towns.api.town.Town
import org.spongepowered.api.Sponge
import org.spongepowered.api.world.Location
import org.spongepowered.api.world.World

interface ClaimService {

    companion object {
        @JvmStatic
        fun get(): ClaimService =
            Sponge.getServiceManager().provideUnchecked(ClaimService::class.java)
    }

    fun getClaimAt(location: Location<World>): Claim?

    fun getClaims(world: World): Collection<Claim>

    fun getClaimsFor(town: Town): Collection<Claim>

    fun register(claim: Claim): Boolean

    fun unregister(claim: Claim): Boolean
}